examples/HowTo.md

```diff
@@ -0,0 +1,147 @@
+
+To add a new logger for your needs you'll have to consider several things.
+Firstly, you should consider choosing some reasonable name for the logger you creating according to the actions that it would cover with logging. From now on let's assume that you need to have `new-logger` in your project.
+
+### Configuration file
+
+To configure your logger you will need configuration `.yaml` file.
+In our example we will use very simple one with the minimal settings needed for the file to work:
+
+```yaml
+loggerTree:
+    severity: Warning+    # severities for «root» logger
+    new-logger:           # logger named «new-logger»
+        severity: Debug+  # severities for logger «new-logger»
+```
+
+For full information about all configuration file's features and options please see [this section](#link-to-readme-config).
+
+### Setting up
+
+Before using your logger you need to set it up. To do that there are very useful functions in `log-warper`.
+In our example we will run all functions with our logger from `main` function of `Main` module.
+
+Let's consider that this is what we have:
+
+```haskell
+mainWithoutLogging :: IO ()
+mainWithoutLogging = inputLength
+```
+ where `inputLength` is the function where we will add some logging messages.
+
+So to make it possible we should transformed this function to the following:
+
+```haskell
+mainWithLogging :: IO ()
+mainWithLogging = do
+    buildAndSetupYamlLogging productionB "examples/how-to-log-config.yaml"
+    usingLoggerName "new-logger" inputLengthWithLog
+```
+where we set up the config from the file and run `inputLengthWithLog` under the logger with name `new-logger`.
+
+Let's have a closer look at how we transform the `inputLength` function to the `inputLengthWithLog` that would work with logging in the next paragraph.
+
+## Add logging messages to function in `IO`
+
+If you want to use logging in a function that already runs in the `IO` monad, replace the monad with [`WithLoggerIO`](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#t:WithLoggerIO) and `liftIO` all your `IO` actions.
+
+For example, our `inputLength` function works this way:
+```haskell
+
+inputLength :: IO ()
+inputLength = do
+    input <- TIO.getLine
+    case T.length input of
+        0 -> putStrLn "Should not be empty" >> inputLength
+          -- v ^  you need to log these
+        _ -> putStrLn "Well done" >> funWithInput input
+
+```
+Very easy, nothing interesting. We have some input from console and write some messages depending of the length of user's input.
+What we are going to do is to log this messages instead of just `putStrLn` them.
+
+First that we should consider is that logging can not be done in `IO`, so we need make the function work in `WithLoggerIO` monad,
```

Comment:
`WithLoggerIO` is not a monad, because monad is some specific type. It's better to say that function should work in some polymorphic monad, which implements `WithLoggerIO` constraint.